dotCMS/src/main/java/com/liferay/portal/struts/PortletRequestProcessor.java

```diff
@@ -330,7 +332,10 @@ protected void doInclude(
 		PortletRequestDispatcherImpl prd = (PortletRequestDispatcherImpl)
 			portletConfig.getPortletContext().getRequestDispatcher(
 				Constants.TEXT_HTML_DIR + uri);
-
+		final User user = WebAPILocator.getUserWebAPI().getLoggedInUser(req);

```

Comment:
you can include a unit test to validate the PageMode depending on the user set. You might also need to mock some stuff

Reply:
@wezell ready. Using the `PortalUtil` class now.